src/compiler/scala/tools/nsc/settings/ScalaSettings.scala

```diff
@@ -55,7 +55,7 @@ trait ScalaSettings extends AbsScalaSettings
   // argfiles is only for the help message
   /*val argfiles = */ BooleanSetting    ("@<file>", "A text file containing compiler arguments (options and source files)")
   val classpath     = PathSetting       ("-classpath", "Specify where to find user class files.", defaultClasspath) withAbbreviation "-cp"
-  val d             = OutputSetting     (outputDirs, ".")
+  val d             = OutputSetting     (outputDirs, ".", "Destination for generated classfiles.")
```

Comment:
True, my reasoning here was to make a generic approach where any overridden setting in the hierarchy would set the new setting value. For example if there is a need to change setting "-x" for scaladoc or any other settings inheriting from scalac settings, all that is needed is:

```scala
override val x = StringSetting(...)
```
With method approach we would have to change the base settings by adding a new method every time.

If the performance/memory is a concern for creating the extra OutputSetting I can make it a method instead. Let me know what you think.